File: ios/Utils/MediaCache.m
#import "MediaCache.h"

@implementation MediaCache
NSMutableArray<NSString *> *completedImagePaths;

+ (void)addCompletedImagePath:(NSString *)imagePath {
    if (imagePath) {
        [completedImagePaths addObject:imagePath];
    }
}

+ (void)removeCompletedImagePath:(NSString *)imagePath {
    if (imagePath) {
        [completedImagePaths removeObject:imagePath];
        NSURL *fileURL = [NSURL URLWithString:imagePath];
        NSString *fileSystemPath = [fileURL path];
        
        NSFileManager *fileManager = [NSFileManager defaultManager];
        NSError *error = nil;
        
        if ([fileManager fileExistsAtPath:fileSystemPath isDirectory:nil]) {
            [fileManager removeItemAtPath:fileSystemPath error:&error];
        }
        
        if (error) {
            NSLog(@"Error deleting image at path: %@", fileSystemPath);
        }
    }
}

+ (void)cleanupCache {
    NSFileManager *fileManager = [NSFileManager defaultManager];
    NSString *cacheDirectory = [NSSearchPathForDirectoriesInDomains(NSCachesDirectory, NSUserDomainMask, YES) firstObject];
    
    for (NSString *imagePath in completedImagePaths) {
        NSString *absoluteImagePath = [cacheDirectory stringByAppendingPathComponent:imagePath];
        NSError *error = nil;
        
        if ([fileManager fileExistsAtPath:absoluteImagePath]) {
            [fileManager removeItemAtPath:absoluteImagePath error:&error];
        }
        
        if (error) {
            NSLog(@"Error deleting image at path: %@", absoluteImagePath);
        }
    }
    
    [completedImagePaths removeAllObjects];
}

@end

File: ios/Utils/MediaCache.swift
import Foundation

class MediaCache {
    static var completedImagePaths: [String] = []

    class func addCompletedImagePath(_ imagePath: String) {
        if !imagePath.isEmpty {
            completedImagePaths.append(imagePath)
        }
    }

    class func removeCompletedImagePath(_ imagePath: String) {
        if !imagePath.isEmpty {
            if let index = completedImagePaths.firstIndex(of: imagePath) {
                completedImagePaths.remove(at: index)
                if let fileURL = URL(string: imagePath) {
                    let fileSystemPath = fileURL.path
                    
                    let fileManager = FileManager.default
                    do {
                        if fileManager.fileExists(atPath: fileSystemPath) {
                            try fileManager.removeItem(atPath: fileSystemPath)
                        }
                    } catch {
                        print("Error deleting image at path: \(fileSystemPath)")
                    }
                }
            }
        }
    }

    class func cleanupCache() {
        let fileManager = FileManager.default
        if let cacheDirectory = NSSearchPathForDirectoriesInDomains(.cachesDirectory, .userDomainMask, true).first {
            for imagePath in completedImagePaths {
                let absoluteImagePath = (cacheDirectory as NSString).appendingPathComponent(imagePath)
                do {
                    if fileManager.fileExists(atPath: absoluteImagePath) {
                        try fileManager.removeItem(atPath: absoluteImagePath)
                    }
                } catch {
                    print("Error deleting image at path: \(absoluteImagePath)")
                }
            }
            completedImagePaths.removeAll()
        }
    }
    
    class func deleteFile(atPath filePath: String) {
        let fileManager = FileManager.default
        var _filePath = filePath
        if _filePath.hasPrefix("file://") {
            _filePath = _filePath.replacingOccurrences(of: "file://", with: "")
       }
        
        if fileManager.fileExists(atPath: _filePath) {
            do {
                try fileManager.removeItem(atPath: _filePath)
                print("File deleted successfully: \(_filePath)")
            } catch {
                print("Error deleting file: \(error.localizedDescription)")
            }
        } else {
            print("File not found at path: \(_filePath)")
        }
    }

}
